docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,251 @@ Note that:
   for instance `tedge/configuration_change//etc/tedge/mosquitto-conf/c8y-bridge.conf`.
 * Since the type of configuration file is used as an MQTT topic name, the characters `#` and `+` cannot be used in a type name.
   If such a character is used in a type name (or in the path of a configuration file without explicit type),
-  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
\ No newline at end of file
+  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+* Notifications for external child-devices are handled differently, as configuration files are fetched/pushed from/to external devices via HTTP file transfer. Thereby notifications are part of the file transfer contract (details see section [Fetching/Pushing configuration file](#fetchingpushing-configuration-file-fromto-the-external-device) below).
+
+## Configuration files for child devices
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` aspects as below are relevant:
+
+* Associating configurations with cloud's child-device twin
+  * an child-device declares its supported config-types to the plugin via MQTT, containing a child-id as reference to the child-device twin
+* Consuming/Providing files from/to thin-edge via network
+  * any configuration file exchange to/from the external child device is handled by an HTTP based file transfer service provided by thin-edge
+
+## Managing Supported Configuration List of child devices
+
+* The `c8y_configuration_plugin` stores and reads the list of supported configuration files for each child-device from a TOML file; similar to whats described in section [Configuration](#configuration) above. Thereby for each child-device an individual TOML file is used.
+* A child-device can also announce its supported configuration files itself to the `c8y_configuration_plugin` via MQTT. The `c8y_configuration_plugin` uses the information received via MQTT to store it into the child-devices individual TOML file.
+
+Each child-devices configuration TOML file is expected in a subfolder, where the subfolder name is treated as `childid`.
+
+Example:
+
+```
+$ tree /etc/tedge/c8y/
+/etc/tedge/c8y/
+├── c8y-configuration-plugin.toml
+├── child1
+│   └── c8y-configuration-plugin.toml
+└── child2
+    └── c8y-configuration-plugin.toml
+```
+
+In that example the plugin serves configuration management for the thin-edge device it-self, for a child-device with childid `child1` and for a child-device with childid `child2`. 
+
+Similar to section [Configuration](#configuration) above, each child-device configuration file is defined by a record with:
+
+* The full `path` to the file. That path is treated as path pointing into the remote filesystem of the child-device.
```

Comment:
> pointing into the remote filesystem of the child-device

Not quite true for a local child device agent as the path would be something on the thin-edge file system itself.

Reply:
I improved that wording. See https://github.com/cstoidner/thin-edge.io/commit/d3259f816b4ad1947a60c6382bcf22d4fa7be720